### PR TITLE
ENH: Update scripts for building Python 3.11 packages

### PR DESCRIPTION
**Pull request by tbirdso**

### CODE REVIEW

**Comment by tbirdso** [comment]
Currently building on `misty`

**Review by dzenanz — APPROVED** [review]
Code looks good - I did not try it.

File: scripts/internal/manylinux-build-common.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ if [[ $# -eq 0 ]]; then
   PYBIN=(/opt/python/*/bin)
   PYBINARIES=()
   for version in "${PYBIN[@]}"; do
-    if [[ ${version} == *"cp37"* || ${version} == *"cp38"* || ${version} == *"cp39"* || ${version} == *"cp310"* ]]; then
+    if [[ ${version} == *"cp37"* || ${version} == *"cp38"* || ${version} == *"cp39"* || ${version} == *"cp310"* || ${version} == *"cp311" ]]; then
```
Comment (thewtex) [inline scripts/internal/manylinux-build-common.sh:15]: Note: #212 likely also brings in new CPython 3.11 into the docker images

File: scripts/internal/windows_build_common.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import os
 import shutil
 
-DEFAULT_PY_ENVS = ["37-x64", "38-x64", "39-x64", "310-x64"]
```
Comment (thewtex) [inline scripts/internal/windows_build_common.py:7]: Will require update to add 3.11 similar to https://github.com/scikit-build/scikit-ci-addons/pull/95/files

**Review by thewtex — COMMENTED** [review]
looking good :+1: 

A few notes inline

File: scripts/internal/windows_build_common.py (hunk 2 above)
Comment (tbirdso) [reply scripts/internal/windows_build_common.py:7]: Added https://github.com/scikit-build/scikit-ci-addons/pull/101 to support.

**Comment by tbirdso** [comment]
See https://github.com/InsightSoftwareConsortium/ITKPythonPackage/pull/216 for discussion of successful MacOS Python 3.11 build after running the install script with updates from this commit.

**Comment by tbirdso** [comment]
Notes:
- Currently building only 3.11 wheels on `blaster`
- Started building 3.11 wheels on my local Windows machine, had to stop halfway due to eating up system resources, can resume overnight. Was able to install Python 3.11 using the script in https://github.com/scikit-build/scikit-ci-addons/pull/101 without issue. Build appeared to be running without issue.

I will update once I have successfully built and imported a Linux 3.11 wheel. @thewtex are there other criteria you can think of for these updates?

File: scripts/internal/windows_build_common.py (hunk 2 above)
Comment (tbirdso) [reply scripts/internal/windows_build_common.py:7]: https://github.com/scikit-build/scikit-ci-addons/pull/101 is merged, thank you @jcfr for your help 👍

**Comment by tbirdso** [comment]
Built Python 3.11 wheels on Linux and was able to import ITK modules successfully in the Python 3.11 console.

Note that the Python 3.11 version included in the docker image is a prerelease from June 2022. I am uncertain whether this is cause for concern.
```bash
[kitware:/work] $ python_311=/opt/python/cp311-cp311/bin/python
[kitware:/work] $ $python_311 --version
Python 3.11.0b3
```

File: scripts/internal/manylinux-build-common.sh (hunk 1 above)
Comment (tbirdso) [reply scripts/internal/manylinux-build-common.sh:15]: @thewtex As noted below the updated docker image does include Python 3.11 but it is a prerelease from a few months before the actual 3.11 release. Any thoughts on whether this needs to be updated in the near term?

**Comment by jcfr** [comment]
> Note that the Python 3.11 version included in the docker image is a prerelease from June 2022. I am uncertain whether this is cause for concern

Looking at the Python release notes [^1], it is indicated:

```
Our goal is have no ABI changes after beta 4 and as few code changes as possible after 3.11.0rc1, the first release candidate.
```

Considering that the version currently in the docker image is `3.11.0b3`, it may be sensible to update the docker image.

A more careful review of changes introduced in a follow-up beta and rc, may help provide a definite answer. That said, to minimize the chance of unintended issues ... I suggest to update. 

[^1]: https://www.python.org/downloads/release/python-3110b3/

**Comment by tbirdso** [comment]
@jcfr After checking it seems we are using the most recent manylinux image: https://hub.docker.com/r/dockcross/manylinux_2_28-x64/tags

I am not familiar with the process to update and tag a new dockcross image with an updated Python version. What steps need to be taken to address the issue?

File: scripts/internal/manylinux-build-common.sh (hunk 1 above)
Comment (thewtex) [reply scripts/internal/manylinux-build-common.sh:15]: Pulling the latest manylinux images, it looks like they now have 3.11.0 -- I will work on pushing an updated image.

**Comment by thewtex** [comment]
> I am not familiar with the process to update and tag a new dockcross image with an updated Python version. What steps need to be taken to address the issue?

dockcross CI usually pushes updates, but there appears to be an error preventing push -- I will build / push locally.

**Review by thewtex — APPROVED** [review]
:rocket:

File: scripts/internal/manylinux-build-common.sh (hunk 1 above)
Comment (tbirdso) [reply scripts/internal/manylinux-build-common.sh:15]: Thank you!

**Comment by thewtex** [comment]
Docker images updated in #218

**Comment by tbirdso** [comment]
For the record, Windows 3.11 builds appear to build, install, and load correctly.

Windows build available here: https://drive.google.com/file/d/1M0OjaJ8vWtTUP0xrKGJ8CvW-K3kOoOSN/view?usp=share_link

```
C:\Python311-x64>.\python.exe -m pip install c:\p\dist\itk_core-5.3rc4.dev20220929-cp311-cp311-win_amd64.whl
Processing c:\p\dist\itk_core-5.3rc4.dev20220929-cp311-cp311-win_amd64.whl
Collecting numpy
  Using cached numpy-1.23.4-cp311-cp311-win_amd64.whl (14.6 MB)
Installing collected packages: numpy, itk-core
  WARNING: The script f2py.exe is installed in 'C:\Python311-x64\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed itk-core-5.3rc4.dev20220929 numpy-1.23.4

C:\Python311-x64> .\python.exe
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import itk
>>> itk.auto_progress(2)
>>> itk.Image
Loading ITKPyBase... done
Loading ITKCommon... done
<itkTemplate itk::Image>
```

**Pull request merged**